🌠Feature/security

beargame123 opened the pull request:
User 작성
UserRepository 작성
AuthUserDetail 작성
FilterConfigure 작성
JwtAuthenticationEntryPoint 작성
JwtTokenProvider 작성
SecurityConfig 작성
UserDetailService 작성

beargame123 committed:
Add files via upload

beargame123 committed:
feature/security

📑 security

dohyeon5626 commented on domain/entity/User.java:
```diff
@@ -0,0 +1,32 @@
+    @Column(length = 100, nullable = false)
+    private String email;
+
+}
```
https://www.erdcloud.com/d/yWjwK2Y4dm52tHdEq
Entity와 데이터베이스에서 설계한 내용이 같아야 할 것 같아요 (User는 id, oauth_id, oauth_type, nickname, introduce, mbti가 있는데 현재 만들어진 내용이랑 다른 것 같네요)

dohyeon5626 commented on domain/repository/UserRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
JpaRepository를 상속받으면 자동으로 빈으로 등록되는데 @Repository를 단 이유가 무엇일까요?

dohyeon5626 commented on global/error/InvalidTokenException.java:
```diff
@@ -0,0 +1,6 @@
+package com.example.mbtogether.global.error;
+
+public class InvalidTokenException extends Throwable {
+    public InvalidTokenException(String token) {
```
매개변수 token이 쓰는 곳이 없는데 왜 입력받는 건가요?
그리고 Throwable말고 RuntimeException을 상속받는 게 더 좋아보입니다.

dohyeon5626 commented on global/security/JwtAuthenticationEntryPoint.java:
```diff
@@ -0,0 +1,26 @@
+        log.error("Responding with unauthorized error. Message - {}", authException.getMessage());
+
+    }
+}
```
AuthenticationEntryPoint을 구현한 이유가 log로 남기기 위해서인가요?
반환되는 내용도 따로 설정하면 더 좋을 것 같습니다.

dohyeon5626 commented on global/security/JwtTokenProvider.java:
```diff
@@ -0,0 +1,93 @@
+
+    private final UserDetailService userLoadService;
+
+    private String secretKey;
```
secretKey는 yml로 받아오는 것이 좋은 것 같습니다.

joungeun replied:
`@Value` 애노테이션을 사용해보세요 [이 글](https://sas-study.tistory.com/273) 참고하시면 될 것 같습니다

jongeuni replied:
`@Value` 애노테이션을 사용해보세요 [이 글](https://sas-study.tistory.com/273) 참고하시면 될 것 같습니다

dohyeon5626 commented on global/security/UserDetailService.java:
```diff
@@ -0,0 +1,28 @@
+import com.example.mbtogether.global.security.detail.AuthUserDetail;
+import com.example.mbtogether.domain.entity.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
```
ADMIN이랑 USER 권한이 있으니 UserDetails는 따로 구현하는 게 좋을 것 같습니다.

joungeun commented on global/security/JwtTokenProvider.java:
```diff
@@ -0,0 +1,93 @@
+
+    private final UserDetailService userLoadService;
+
+    private String secretKey;
```
@Value 애노테이션을 사용해보세요 [이 글](https://sas-study.tistory.com/273) 참고하시면 될 것 같습니다

dohyeon5626 commented on src/main/java/com/example/mbtogether/config/SecurityConfig.java:
```diff
@@ -0,0 +1,29 @@
+                .antMatchers("/admin/**").hasRole("ADMIN")
+                .antMatchers("/**").permitAll()
+                    .and()
+                .formLogin()
```
formLogin을 하지 않으니 굳이 로그인 url을 설정하고 성공 페이지를 설정할 필요 없을 것 같아요.
일반적인 API에 따른 권한 설정만 하면 될 것 같습니다.

beargame123 committed:
feature/security

📑User 작성
📑UserRepository 작성
📑AuthUserDetail 작성
📑FilterConfigure 작성
📑JwtAuthenticationEntryPoint 작성
📑JwtTokenProvider 작성
📑SecurityConfig 작성
📑UserDetailService 작성

beargame123 committed:
feature/security

beargame123 committed:
Add files via upload
